core/src/main/scala/akka/persistence/r2dbc/internal/R2dbcExecutor.scala

```diff
@@ -199,6 +237,12 @@ class R2dbcExecutor(val connectionFactory: ConnectionFactory, log: Logger)(impli
     }
   }
 
+  def updateInBatch(logPrefix: String)(statement: Connection => Statement): Future[Int] =
+    withConnection(logPrefix) { connection =>
```

Comment:
Good, as far as I can see we can change the `update` method to also use `withConnection`. The only difference would be the debug logging of number of rows updated, but that is not important. Could you change that at the same?

Reply:
yes, that's what I was planning to change on top of this PR, but didn't want to add too much here. 

Also I think could have two methods. The `withConnection` is more about a connection with a open transaction. 
So maybe we should have `withConnection` using auto-comit and `withConnectionTx` for the transactional one. Or some other naming.